tsl/src/compression/api.c

```diff
@@ -199,18 +280,138 @@ restore_autovacuum_on_decompress(Oid uncompressed_hypertable_relid, Oid uncompre
 	}
 }
 
-static void
+static Chunk *
+find_chunk_to_merge_into(Hypertable *ht, Chunk *current_chunk)
+{
+	int64 max_chunk_interval, current_chunk_interval = 0, compressed_chunk_interval = 0;
+	const Dimension *time_dim;
+	Chunk *previous_chunk;
+	Point *p;
+
+	time_dim = hyperspace_get_open_dimension(ht->space, 0);
+	if (!time_dim)
+		elog(ERROR, "hypertable has no open partitioning dimension");
+
+	if (time_dim->fd.compress_interval_length == 0)
+		return NULL;
+
+	if (current_chunk->cube->num_slices == 0 ||
+		current_chunk->cube->slices[0]->fd.dimension_id != time_dim->fd.id)
+	{
+		elog(ERROR, "primary dimension of the chunk is not the time dimension");
+	}
```

Comment:
```suggestion
	Assert(current_chunk->cube->slices[0]->fd.dimension_id == time_dim->fd.id);
```
You cannot create a hypertable with with time not being primary dimension with our API

Reply:
I wasn't able to find a way to create a hypertable without primary open dimension. Definitely not possible through the API.